pkg/common/utils/utils_test.go

```diff
@@ -331,6 +331,44 @@ func TestGetNamespaceGuaranteedFromAnnotation(t *testing.T) {
 	}
 }
 
+func TestGetNamespaceMaxAppsFromAnnotation(t *testing.T) {
+	testCases := []struct {
+		namespace      *v1.Namespace
+		expectedMaxApp string
+	}{
+		{&v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "test",
+			},
+		}, ""},
+		{&v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "test",
+				Annotations: map[string]string{
+					constants.NamespaceMaxApps: "5",
```

Comment:
Could you add a negative test case as well? eg. `-5`.

Reply:
Good catch, addressed in latest PR!